scripts/jenkins-yoctobuild-build.sh

```diff
@@ -397,14 +402,17 @@ fi
 # private docker containers, require login:
 docker login -u menderbuildsystem -p ${DOCKER_PASSWORD}
 
-# if we abort a build, docker might still be up and running
-docker ps -q -a | xargs -r docker stop || true
-docker ps -q -a | xargs -r docker rm -f || true
-sudo chmod 777 /var/run/docker.sock
+# Cleanup specific for Jenkins reusable VMs
+if [ -n "$JENKINS_URL" ]; then
+    # if we abort a build, docker might still be up and running
+    docker ps -q -a | xargs -r docker stop || true
+    docker ps -q -a | xargs -r docker rm -f || true
+    sudo chmod 777 /var/run/docker.sock
```

Comment:
Are Gitlab nodes reused? Because some step similar to this is definitely needed if they are. But I guess we should not kill the runner... :-)

Reply:
We do reuse the containers, actually. It is something I was not expecting but I have experienced.

I will add for GitLab runners steps: stop, rm, prune and killi the stress client; while leaving Jenkins only the chmod of the socket and the restart. Do you agree?

Reply:
Hmm, is that correct..? Ping me when you see this.

Reply:
Discussed in person and verified manually. As we are using DinD design, we cannot see containers that we have not started ourselves, so it is safe to call stop, rm, etc.

Reply:
Smart solution, that DinD!